check_patroni/types.py

```diff
@@ -1,19 +1,28 @@
 import logging
-import urllib3
 
 import attr
 import nagiosplugin
-from typing import Any, Callable, List
+import requests
+import urllib3
+from typing import Any, Callable, List, Optional, Tuple, Union
 
 _log = logging.getLogger("nagiosplugin")
 
 
+class APIError(requests.exceptions.RequestException):
+    """This exception is raised when the rest api couldn't
+    be reached and we got a http status code different from 200.
+    """
+
+    pass
```

Comment:
The inheritance is not really appropriate I think; just subclass `Exception`.

Also, the `pass` statement is superfluous.